mirrord/kube/src/api/container/job.rs

```diff
@@ -83,15 +83,67 @@ where
 
     let mut pod_progress = progress.subtask("waiting for pod to be ready...");
 
-    while let Some(Ok(pod)) = stream.next().await {
-        let Some(phase) = pod.status.as_ref().and_then(|status| status.phase.as_ref()) else {
-            continue;
-        };
-
-        debug!(?phase, "Agent pod changed");
-
-        if phase == "Running" {
-            break;
+    // We want to warn the user if the pod has been in the "Pending" phase for 10+ seconds
+    let mut is_pending = false;
+    let mut warned_user = false;
+    let pending_timer = sleep(Duration::from_secs(10));
+    tokio::pin!(pending_timer);
+
+    loop {
+        tokio::select! {
+            _ = &mut pending_timer, if is_pending && !warned_user => {
```

Comment:
```suggestion
            _ = &mut pending_timer, if is_pending && !warned_user => {
```
I think in the final version, this loop will aways exit when the pod exits the `Pending` phase. So no need to store `is_pending`